CodeCamp #139 [Feature] Support REFUGE dataset. (#2420)
open-mmlab/mmsegmentation: docs/en/user_guides/2_dataset_prepare.md

The script will make directory structure automatically.

### REFUGE

> First, registrate in [REFUGE2020Challenge](https://refuge.grand-challenge.org)and download data from [this link](https://refuge.grand-challenge.org/REFUGE2Download)

Reviewer: We may tell users what raw data folder directory looks like in `raw_data_root`.

> To convert REFUGE dataset to MMSegmentation format, you should run the following command:

Reviewer: For example, in [Synapes](https://github.com/open-mmlab/mmsegmentation/blob/dev-1.x/docs/en/user_guides/2_dataset_prepare.md#synapse-dataset) dataset, it shows:

```
│   ├── iSAID
│   │   ├── train
│   │   │   ├── images
│   │   │   │   ├── part1.zip
│   │   │   │   ├── part2.zip
│   │   │   │   ├── part3.zip
│   │   │   ├── Semantic_masks
│   │   │   │   ├── images.zip
│   │   ├── val
│   │   │   ├── images
│   │   │   │   ├── part1.zip
│   │   │   ├── Semantic_masks
│   │   │   │   ├── images.zip
│   │   ├── test
│   │   │   ├── images
│   │   │   │   ├── part1.zip
│   │   │   │   ├── part2.zip
```

```shell

> python tools/convert_datasets/refuge.py --raw_data_root=.. --tmp_dir=..

Reviewer: Make `REFUGE.zip` which could be downloaded from official website as `raw_data` thus users could generate mmsegmentation-style data automatically.

python tools/dataset_converters/synapse.py --dataset-path /path/to/synapse
```

Noted that MMSegmentation default evaluation metric (such as mean dice value) is calculated on 2D slice image,
which is not comparable to results of 3D scan in some paper such as [TransUNet](https://arxiv.org/abs/2102.04306).

### REFUGE

First, registrate in [REFUGE2020Challenge](https://refuge.grand-challenge.org)and download data from [this link](https://refuge.grand-challenge.org/REFUGE2Download)

Then, unzip `REFUGE2.zip` and the contents of original datasets include:

```none
├── REFUGE2
│   ├── REFUGE2
│   │   ├── Annotation-Training400.zip
│   │   ├── REFUGE-Test400.zip
│   │   ├── REFUGE-Test-GT.zip
│   │   ├── REFUGE-Training400.zip
│   │   ├── REFUGE-Validation400.zip
│   │   ├── REFUGE-Validation400-GT.zip
│   ├── __MACOSX
```

> To convert REFUGE dataset to MMSegmentation format, you should run the following command:

Reviewer 2: ```suggestion
Please run the following command to convert REFUGE dataset:
```

```shell
python tools/convert_datasets/refuge.py --raw_data_root=/path/to/refuge/REFUGE2/REFUGE2
```

> The script will make directory structure automatically.
> 
> In our default setting, it will generate 400 images for training, 400 images for validation and 400 images for testing.

Reviewer 2: ```suggestion
There is the dataset structure after running `tools/convert_datasets/refuge.py`, which includes 400 images for training, 400 images for validation and 400 images for testing.
```
also add the proper dataset structure example
